### DISCUSSION TITLE
GtkDialog mapped without a transient parent.

### DISCUSSION THREAD

**Discussion by ITwrx** [can-lehmann/owlkettle#50] [Q&A] [answered]
I'm getting:
```
Gtk-CRITICAL **: 13:49:51.217: gtk_window_set_transient_for: assertion 'parent == NULL || GTK_IS_WINDOW (parent)' failed
Gtk-Message: 13:49:51.223: GtkDialog mapped without a transient parent. This is discouraged.
```
when i click on the button that opens my `custom_dialog`. From reading around, it seems `gtk` needs to be told what the `Parent Window` of the `Dialog` is, but i'm not sure if it needs it's immediate upstream `viewable`, or the actual top-level (and currently only) upstream `Window`, and i'm not sure how, or where, to specify either.

My structure is:

`main.nim`: 
which imports `views/loans/list_loans_view`, 
and includes/uses
```
LoanListView:
  loans = app.loan_model.loans
```

`/views/loans/list_loans_view.nim`:
which imports `add_loan_dialog`
and includes/uses:

```
Button {.expand: false.}:
            text = "+ Loan"
            style = {ButtonSuggested}
            proc clicked() =
              let (res, state) = loanListView.open(gui(AddLoanDialog()))
              if res.kind == DialogAccept:
                addNewLoanToLoans(AddLoanDialogState(state).newLoan) 
                loanListView.loans = getAllLoans()
```
`/views/loans/add_loan_dialog` : 
which has the `AddLoanDialog viewable` and associated `view`.
 
thanks

**Comment by can-lehmann** [comment, marked as answer]
> From reading around, it seems gtk needs to be told what the Parent Window of the Dialog is, but i'm not sure if it needs it's immediate upstream viewable, or the actual top-level (and currently only) upstream Window, and i'm not sure how, or where, to specify either.

You are correct, GTK needs the "parent window" for the dialog. This means that you need to pass the toplevel viewable to the `open` function. This is actually pretty simple to do, since every viewable has the toplevel viewable stored in the `Viewable.app` field. Try using `loanListView.app.open` instead of `loanListView.open`.

**Reply by ITwrx** [reply]
Thanks a lot! That fixed it.
